src/react-medium-image-zoom.js

```diff
@@ -78,11 +78,13 @@ export default class ImageZoom extends Component {
     this.removeZoomed()
   }
 
-  // We need to check to see if any changes are being
-  // mandated by the consumer and if so, update accordingly
+  /**
+   * We need to check to see if any changes are being
+   * mandated by the consumer and if so, update accordingly
+   */
   componentWillReceiveProps(nextProps) {
     const imageChanged = this.props.image.src !== nextProps.image.src
-    const isZoomedChanged = this.props.isZoomed !== nextProps.isZoomed
+    const isZoomedChanged = this.state.isZoomed !== nextProps.isZoomed
```

Comment:
This needed to change to `state` because the internal state of being open/closed can change without the external props changing. Basically, this always checks against and gives preference to `nextProps.isZoomed`